commands/social/store/store.js

```diff
@@ -106,26 +69,109 @@ class Store extends Social {
+    }
+    const filter = m => m.author.id === message.author.id;
+    const response = await client.awaitReply(message, `Are you sure you want to purchase a Slot Token for ₲${message.settings.tokenPrice}?`, filter, undefined, null);
+    if (["y", "yes"].includes(response.toLowerCase())) {
+    
+      message.member.takePoints(Number(message.settings.tokenPrice));
+      await message.member.giveItem("tokens", 1);
+      message.channel.send("You have bought a token");
+    
+    } else
+    
+    if (["n", "no", "cancel"].includes(response.toLowerCase())) {
+      message.response(undefined, "Transaction cancelled.");
+    }
+  }
+  async buyRole(name, message, client) {
+    const item = client.store.filter(i => i.name.toLowerCase().includes(name.toLowerCase()));
+    
+    if (item.size > 1) return message.reply(`B..Baka! Be more specific more than that, there is more than one item on sale with ${name} as their name`);
+    if (!item) return message.channel.send("That item doesn't exist, Please make sure it is spelled correctly");
+    if (message.member.roles.has(item.first().id)) return message.channel.send("You already have the role :facepalm: ");
+    
+    if (Number(item.first().price) > Number(message.member.score.points)) {
+      return message.channel.send(`You currently have ₲${Number(message.member.score.points).toLocaleString()}, but the role costs ${Number(item.first().price).toLocaleString()}!`);
+    }
+  
+    const filter = m => m.author.id === message.author.id;
+    const response = await client.awaitReply(message, `Are you sure you want to purchase ${item.first().name} for ₲${item.first().price.toLocaleString()}?`, filter, undefined, null);
+    if (["y", "yes"].includes(response.toLowerCase())) {
+    
+      message.member.takePoints(Number(item.first().price));
+      await message.member.roles.add(item.first().id);
+      message.channel.send("You have bought the role :tada: ");
+    
+    } else
+    
+    if (["n", "no", "cancel"].includes(response.toLowerCase())) {
+      message.response(undefined, "Transaction cancelled.");
+    }
+  }
+  async sellToken(name, message, client) {
+    const returnPrice = Number(message.settings.tokenPrice) / 2;
+    const filter = m => m.author.id === message.author.id;
+    const response = await client.awaitReply(message, `Are you sure you want to sell a Token for ₲${returnPrice}?`, filter, undefined, null);
+    if (["y", "yes"].includes(response.toLowerCase())) {
+    
+      message.member.takePoints(returnPrice);
+      message.member.takeItem("tokens", 1);
+      message.channel.send("You have sold a token");
+    
+    } else
+    
+    if (["n", "no", "cancel"].includes(response.toLowerCase())) {
+      message.response(undefined, "Transaction cancelled.");
+    }
+  }
+  
+  async sellRole(name, message, client) {
```

Comment:
you shouldn't be passing client, you can still access it with `this.client`... you're in the class